File: app/src/main/java/com/mytlogos/enterprise/background/api/UserApi.java
package com.mytlogos.enterprise.background.api;

import com.mytlogos.enterprise.background.api.model.ClientChangedEntities;
import com.mytlogos.enterprise.background.api.model.ClientDownloadedEpisode;
import com.mytlogos.enterprise.background.api.model.ClientMediaList;
import com.mytlogos.enterprise.background.api.model.ClientNews;
import com.mytlogos.enterprise.background.api.model.ClientStat;
import com.mytlogos.enterprise.background.api.model.ClientToc;
import com.mytlogos.enterprise.background.api.model.ClientUser;
import com.mytlogos.enterprise.background.api.model.InvalidatedData;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

interface UserApi {

    @POST("{start}/logout")
    Call<Boolean> logout(@Path(value = "start", encoded = true) String url, @Body Map<String, Object> body);

    @PUT
    Call<Boolean> updateUser(@Url String url, @Body Map<String, Object> body);

    @GET
    Call<ClientUser> getUser(@Url String url, @QueryMap Map<String, Object> body);

    @GET("{start}/news")
    Call<List<ClientNews>> getNews(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/lists")
    Call<List<ClientMediaList>> getLists(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/invalidated")
    Call<List<InvalidatedData>> getInvalidated(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/download")
    Call<List<ClientDownloadedEpisode>> downloadEpisodes(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/stats")
    Call<ClientStat> getStats(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/new")
    Call<ClientChangedEntities> getNew(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @GET("{start}/toc")
    Call<List<ClientToc>> getToc(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @DELETE("{start}/toc")
    Call<Boolean> removeToc(@Path(value = "start", encoded = true) String url, @QueryMap Map<String, Object> body);

    @POST("{start}/toc")
    Call<Boolean> addToc(@Path(value = "start", encoded = true) String url, @Body Map<String, Object> body);
}

File: app/src/main/java/com/mytlogos/enterprise/background/api/UserApi.kt
package com.mytlogos.enterprise.background.api

import com.mytlogos.enterprise.background.api.model.*
import retrofit2.Response
import retrofit2.http.*

internal interface UserApi {
    @POST("{start}/logout")
    suspend fun  logout(@Path(value = "start", encoded = true) url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>

    @PUT
    suspend fun updateUser(@Url url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>

    @GET
    suspend fun  getUser(@Url url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientUser>

    @GET("{start}/news")
    suspend fun  getNews(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<List<ClientNews>>

    @GET("{start}/lists")
    suspend fun  getLists(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<List<ClientMediaList>>

    @GET("{start}/download")
    suspend fun downloadEpisodes(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<List<ClientDownloadedEpisode>>

    @GET("{start}/stats")
    suspend fun  getStats(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientStat>

    @GET("{start}/new")
    suspend fun  getNew(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<ClientChangedEntities>

    @GET("{start}/toc")
    suspend fun  getToc(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<List<ClientToc>>

    @DELETE("{start}/toc")
    suspend fun  removeToc(@Path(value = "start", encoded = true) url: String, @QueryMap body: MutableMap<String, Any?>): Response<Boolean>

    @POST("{start}/toc")
    suspend fun  addToc(@Path(value = "start", encoded = true) url: String, @Body body: MutableMap<String, Any?>): Response<Boolean>
}